ansible/roles/providers/tasks/aws.yml

```diff
@@ -33,6 +33,36 @@
   when:
     - ansible_distribution == "Flatcar"
 
+
+- name: install cloud-init deps
+  yum:
+    name: "{{ packages }}"
+    state: present
+    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+  vars:
+    packages:
+      - cloud-utils-growpart
+      - "python{{ '3' if ansible_distribution_major_version|int >= 9 else '2'}}-pip"
+  when: ansible_os_family == "RedHat"
+
+- name: install cloud-init for RHEL
+  yum:
+    name: cloud-init
+    state: present
+    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+  when:
+  - ansible_os_family == "RedHat"
+  - ansible_distribution_version is version('8.10', '!=')
+
+ #this pins the version of cloud-init to avoid the issue here https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4745
+- name: Install cloud-init 23.1.1 for RHEL 8.10
+  shell: rpm -Uvh --oldpackage "{{ offline.os_packages_remote_filesystem_repo_path }}/cloud-init-23.1.1-10.el8.noarch.rpm"
```

Comment:
Is `--oldpackage` here because the version is being downgraded?

Reply:
There is also https://docs.ansible.com/ansible/9/collections/ansible/builtin/yum_module.html#parameter-allow_downgrade

Reply:
added it as part of the networked usecase